sentinel/src/schedule/outbound.js

```diff
@@ -0,0 +1,232 @@
+    body: payload,
+    json: true
+  };
+
+  const auth = () => {
+    const authConf = conf.destination.auth;
+
+    if (!authConf) {
+      return Promise.resolve();
+    }
+
+    if (!authConf.type) {
+      return Promise.reject(Error('No auth.type, either declare the type or omit the auth property'));
+    }
+
+    if (authConf.type.toLowerCase() === 'basic') {
+      return credential(authConf['password_key'])
+        .then(password => {
+          sendOptions.auth = {
+            username: authConf.username,
+            password: password,
+            sendImmediately: true
+          };
+        });
+    }
+
+    if (authConf.type.toLowerCase() === 'muso-sih') {
+      return credential(authConf['password_key'])
+        .then(password => {
+          const authOptions = {
+            method: 'POST',
+            form: {
+              login: authConf.username,
+              password: password
+            },
+            url: urlJoin(conf.destination.base_url, authConf.path),
+            json: true
+          };
+
+          return request(authOptions)
+            .then(result => {
+              // No that's not a spelling mistake, this API is sometimes French!
+              if (result.statut !== 200) {
+                logger.error('Non-200 status from Muso auth', result);
+                throw new Error(`Got ${result.statut} when requesting auth`);
+              }
+
+              sendOptions.qs = {
+                token: result.data.username_token
+              };
+            });
+        });
+    }
+
+    // Misconfigured auth type
+    return Promise.reject(new Error(`Invalid auth type '${authConf.type}'. Supported: basic, muso-sih`));
+  };
+
+  return auth().then(() => request(sendOptions));
+};
+
+// Collects pushes to attempt out of the queue, given a global config
+const collect = (config, queue) => {
+  return queue.queue.map(pushName => {
+    return config.find(conf => conf.key === pushName);
+  });
+};
+
+// Coordinates the attempted pushing of documents that need it
+const execute = () => {
+  const pushConfig = arrayinate(configService.get(CONFIGURED_PUSHES) || {});
+  if (!pushConfig.length) {
+    return Promise.resolve();
+  }
+
+  return module.exports._queued()
+  .then(queues => {
+    // array of {doc, conf} to be processed
+    const pushes = queues.reduce((pushes, [queue, doc]) => {
+      const pushesForDoc =
+        module.exports._collect(pushConfig, queue).map(conf => ({queue, doc, conf}));
+      return pushes.concat(pushesForDoc);
+    }, []);
+
+    // Attempts each push one by one. Written to be simple not efficient.
+    // There are lots of things we could do to make this faster / less fragile,
+    // such as scoping pushes by domain, as well as writing out successes before
+    // all pushes are complete
+    // For now we presume we aren't going to get much traffic against this and
+    // will probably only be doing one push per schedule call
+    const dirtyQueues = {};
```

Comment:
It's marking task documents that need to be written back to CouchDB because they have changed. It should hopefully be clearer after your suggested rename. I'm going by the use of dirty as talked about here: https://stackoverflow.com/a/1745899/1666